src/main/sensors/gyro.c

```diff
@@ -531,22 +538,20 @@ FAST_CODE void gyroFiltering(timeUs_t currentTimeUs)
     }
 #endif
 
-    if (!overflowDetected) {
-        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-            gyroFilteredDownsampled[axis] = pt1FilterApply(&gyro.imuGyroFilter[axis], gyro.gyroADCf[axis]);
-        }
-    }
+    // Use clock cycle count at gyro interrupt to get as much precision as possible
+    // use currentTimeUs converted to cycles if interrupts are not used
+    const bool interruptEnabled = gyro.gyroSensor1.gyroDev.gyroModeSPI != GYRO_EXTI_NO_INT;
+    const uint32_t cycleStamp = interruptEnabled ? gyro.gyroSensor1.gyroDev.gyroLastEXTI : clockMicrosToCycles(currentTimeUs);
```

Comment:
(with clockMicrosToCycles, data calculation will fail on overflow. Interval between overflows is large, so it is only theoretical problem)